Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -891,21 +893,35 @@ void MatrixWorkspace::setYUnit(const std::string &newUnit) {
 }
 
 /// Returns a caption for the units of the data in the workspace
-std::string MatrixWorkspace::YUnitLabel() const {
+std::string MatrixWorkspace::YUnitLabel(bool useLatexText /* = false */) const {
   std::string retVal;
-  if (!m_YUnitLabel.empty())
+  if (!m_YUnitLabel.empty()) {
```

Comment:
This case is where a user has specifically specified a unit by hand and I think we should probably just leave that alone as it was. They can set latex labels themselves if they wish but that should probably be their choice.

Reply:
The file mentioned in the issue (#25517 `ExternalData/Testing/SystemTests/tests/analysis/reference/MARIReductionMonSeparate.nxs`) enters this block when plotting i.e. `if (!m_YUnitLabel.empty())` evaluates to true. Which is why I added the Latex conversion here as well (without the conversion the label would be `Counts per meV`). Is this expected?

Reply:
I think I see the issue here. It looks like we save the return value of `yUnitLabel()` in to the `.nxs` and this will never be empty because it's either the user set value or our computed value from the `else` clause. As a result, when we load it back in we get `m_YUnitLabel` not empty which looks like a custom label was set.

This is a tricky one to deal with as we don't save enough information to know whether the label was user set or we computed it. We might have to leave the regex check for now. At least here it is closer to the origin of the `"per"` string.